### ISSUE TITLE
Catkin_make fails to run 

### ISSUE THREAD

**Issue by A-I-Ivanov** [ros-industrial/ros_qtc_plugin#291]
After installing using the .run file and following the import instructions on the wiki, the following error occurs:

 14:15:37: Could not start process "catkin_make" --cmake-args -G 'CodeBlocks - Unix Makefiles' -DCMAKE_BUILD_TYPE=Debug
Error while building/deploying project untrusted_maps_planner (kit: Desktop)
When executing step "CatkinMake Step"

A similar error is present if I try release mode etc. Is this a permissions problem?  Notably, I run qt from command line and the same issue does not occur if I run catkkin_make in command line. Everything builds as expected.

**Comment by Levi-Armstrong** [comment]
If you launch qt creator from a terminal then does it build?

**Comment by A-I-Ivanov** [comment]
If you read my note, I have been launching from the termina. Also, I have
roa sourced in my bashrc.

Best ,
Alex

**Comment by Levi-Armstrong** [comment]
>Notably, I run qt from command line and the same issue does not occur if I run catkkin_make in command line. Everything builds as expected.

The statement above is not clear to me what you actual tried. The way I read it is you launch Qt Creator from the terminal then from a terminal you call catkin_make and everything builds fine. I am asking if you launched Qt Creator from a terminal and then from within Qt Creator does it build?

**Comment by A-I-Ivanov** [comment]
Sorry for the late response. No, I mean I try to build things through Qt alone (having launched it through a terminal with ros sourced). This is unsuccessful. Conversely, simply calling catkin_make in a terminal is successful.

**Comment by Levi-Armstrong** [comment]
Have you compared the environment variables from a terminal where it builds to the ones in Qt Creator>Projects>Build?

**Comment by A-I-Ivanov** [comment]
The variables seemed fine, but then when I tried to build it they changed. In addition, qt made a new build folder called (Defalt-src.....), generated another CMakeLists top file called CMakeLists.txt.user, and also somehow broke catkin. Now the project doesn't build at all. I think I may have to reinstall ros. No offense here, but this has cost me quite a few hours, so I don't know how much further I'll pursue this instead of switching back to my working IDE.

**Comment by Levi-Armstrong** [comment]
If you are getting a CMakeLists.txt.user then the workspace was not imported using the this plugin. This file get created if you open it as a pure cmake project. This also explains why it made a new build folder. I recommend removing any .user file and clean the workspace by removing everything except the src folder. Then follow [this](https://ros-industrial.github.io/ros_qtc_plugin/_source/Import-ROS-Workspace.html) wiki.

**Comment by A-I-Ivanov** [comment]
Hi Levi, I had to reinstall Ubuntu a few days ago, and I took the time to try again. I ended up following the wiki you mentioned, but still no dice. The same error as before. I also installed the default Qt creator by itself (after having followed your install guide for qtcreator-ros and following the wiki). I was able to get qt to build by taking the copying the toplevel cmake file into my Catkin workspace, renaming it to CMakeLists.txt, then directly loading the project from this new file. I'm not sure what the difference is between loading from a CMakeLists.txt file and using your plugin. Thanks for taking the time. If you need more info/screenshots from me, let me know.

**Comment by Levi-Armstrong** [comment]
Thank you for he update. Screenshots would be helpful and if the project is public I could try building it on my end.

**Comment by A-I-Ivanov** [comment]
Levi, I'll try to get screen shots to you this weekend. Unfortunately it's not public at the moment, but I can mirror it on github temporarily for you. Ill get back to you asap, but since this project is on the back burner, it takes me some time to respond. Sorry for the delay.

**Comment by A-I-Ivanov** [comment]
Hi Levi. It seems that the problem is now resolved, but I'm not exactly sure why. When I get done with this project, I will attempt to do everything from scratch and see what happens. Of note, there was a problem where when I sourced my workspace (in command line) it broke roslaunch and my path variables. I then deleted my workspace and re-cloned. Both the roslaunch problem and qtcreator-ros problems resolved simultaneously. I'm sorry I can't give more details at the moment, but it has something to do with borked paths in my estimation. The strange thing is I re-installed Ubuntu and did everything fresh and the problem persisted until just now. Another thing of note is that I installed a newer version of qt-creator after using your installation instructions (and problems precising). It seems that the newer version of qt works just fine with your plugin.
